presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -0,0 +1,740 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.type;
```

Comment:
Yeah, we can figure out how/where to move it later.

BTW, one thing I've been thinking about is that other than a few factory methods and conversion methods, this class doesn't have much that is specific to "decimal". We might want to split it into a `UnsignedInt128` class that just knows how to perform the primitive operations and move the decimal-specific methods to the `Decimals` class. We can do that refactor later, though.